x-pack/plugins/fleet/public/components/agent_enrollment_flyout/agent_policy_selection.tsx

```diff
@@ -38,23 +58,28 @@ const resolveAgentId = (
     if (agentPolicies.length === 1) {
       return agentPolicies[0].id;
     }
-
-    const defaultAgentPolicy = agentPolicies.find((agentPolicy) => agentPolicy.is_default);
-    if (defaultAgentPolicy) {
-      return defaultAgentPolicy.id;
-    }
   }
 
   return selectedAgentPolicyId;
 };
 
 export const EnrollmentStepAgentPolicy: React.FC<Props> = (props) => {
-  const { agentPolicies, onAgentPolicyChange, excludeFleetServer } = props;
+  const {
+    agentPolicies,
+    onAgentPolicyChange,
+    excludeFleetServer,
+    onClickCreatePolicy,
+    selectedAgentPolicy,
+    isFleetServerPolicy,
+  } = props;
 
   const [selectedAgentPolicyId, setSelectedAgentPolicyId] = useState<undefined | string>(
     () => resolveAgentId(agentPolicies, undefined) // no agent id selected yet
   );
 
+  // Create new agent policy flyout state
+  const hasWriteCapabilites = useCapabilities().write;
```

Comment:
This will need to be updated to use the new authz object instead (see #122347). Depending on which gets merged first, we'll need to make sure this gets updated too. FYI @criamico

Reply:
@criamico @joshdover I updated here (create agent policy) to use `authz.fleet.all` like it was on create agent policy flyout before, please verify it is correct:
previously: https://github.com/elastic/kibana/pull/122347/files#diff-4afa7a3dae1c92ca6a869091d97de65096eb11cb20dcd0b3b7925263704c077b
updated: https://github.com/elastic/kibana/pull/121628/commits/511c236206c926ba5bf83871883872569a9aca05

Reply:
It looks good to me, I don't think other privileges are needed in this case.